src/components/Assets/AssetsList.tsx

```diff
@@ -317,9 +319,54 @@ const AssetsList = () => {
       breadcrumbs={false}
       hideBack
       options={
-        <>
+        <div className="flex flex-wrap items-center gap-3">
+          <div className="mb-2 flex w-full flex-col items-center gap-3 lg:mb-0 lg:w-fit lg:flex-row">
+            <div className="w-full lg:w-fit">
+              <Button
+                variant="primary"
+                size="default"
+                className="w-full p-[10px] md:w-auto"
+                onClick={() => setIsScannerActive(true)}
+              >
+                <CareIcon icon="l-search" className="text-base mr-2" />
+                Scan Asset QR
+              </Button>
+            </div>
+
+            <div className="w-full lg:w-fit">
+              <Button
+                variant="primary"
+                size="default"
+                id="create-asset-button"
+                disabled={!NonReadOnlyUsers}
+                className="w-full p-[10px] md:w-auto"
+                onClick={() => {
+                  if (qParams.facility) {
+                    navigate(`/facility/${qParams.facility}/assets/new`);
+                  } else {
+                    setShowFacilityDialog(true);
+                  }
+                }}
+              >
+                <CareIcon icon="l-plus-circle" className="text-lg mr-2" />
+                <span>{t("create_asset")}</span>
+              </Button>
+            </div>
+          </div>
+
+          {/* Advanced Filter Button */}
+          <div className="w-full md:w-auto">
+            <AdvancedFilterButton
```

Comment:
let's rename the component too.

```suggestion
            <FilterButton
```